saq/queue.py

```diff
@@ -529,7 +541,10 @@ async def enqueue(self, job_or_func: str | Job, **kwargs: t.Any) -> Job | None:
             ):
                 return None
 
-        logger.info("Enqueuing %s", job)
+        logger.info(
+            "Enqueuing %s",
+            repr(job) if logger.isEnabledFor(logging.DEBUG) else str(job),
```

Comment:
maybe make this a helper function because it's very repetitive. i don't like how repr and str are different, perhaps it could be an argument to the helper function for simple or verbose logging

Reply:
That's a good point. Unfortunately can't add arguments to repr or str, but a helper function, sure.

Reply:
I created an `info()` method that takes an optional boolean for `full`. This now simplifies to `job.info(logger.isEnabledFor(logging.DEBUG))` which is still more verbose than I would like, but we don't know the status of the logger at that point unless we pass it in.
And I decided that it was better to keep the `info()` function's behaviour independent of global state.